src/com/mendix/widget/GoogleMaps/components/__tests__/Map.spec.ts

```diff
@@ -0,0 +1,201 @@
+import { shallow } from "enzyme";
+import GoogleMap from "google-map-react";
+import { DOM, createElement } from "react";
+
+import { Map, MapProps } from "../Map";
+import { Marker } from "../Marker";
+
+import {
+    EventMock, GeocoderLocationType, GeocoderMock, GeocoderStatus, LatLngBoundsMock,
+    LatLngMock, MapsMock, MarkerMock, MockGoogle
+} from "../../../../../../../tests/mocks/GoogleMaps";
+
+import { MxMock, MxUiMock } from "../../../../../../../tests/mocks/Mendix";
+describe("Map", () => {
+    const address = "Lumumba Ave, Kampala, Uganda";
+    const invalidAddress = "invalidAddress";
+    const APIKey = "AIzaSyACjBNesZXeRFx86N7RMCWiTQP5GT_jDec";
+    const renderMap = (props: MapProps) => shallow(createElement(Map, props));
+    const defaultCenterLocation = { lat: 51.9107963, lng: 4.4789878 };
+    const successMockLocation = { lat: 30, lng: 118 };
+    const multipleAddressMockLocation = { lat: 34.213171, lng: -118.571022 };
+
+    beforeAll(() => {
+        window.google = MockGoogle;
+        window.google.maps = {};
+        window.google.maps.Geocoder = GeocoderMock;
+        window.google.maps.Map = MapsMock;
+        window.google.maps.LatLngBounds = LatLngBoundsMock;
+        window.google.maps.LatLng = LatLngMock;
+        window.google.maps.Marker = MarkerMock;
+        window.google.maps.event = EventMock;
+        window.google.maps.GeocoderStatus = GeocoderStatus;
+        window.google.maps.GeocoderLocationType = GeocoderLocationType;
+        window.mx = MxMock.prototype;
+        window.mx.ui = MxUiMock.prototype;
```

Comment:
You need to define only one mocks here: `window.mx = { ui: MxUiMock };`.

Value for `google.maps.Geocoder` and for `google.maps.GeocoderStatus` you should set only when you emulate `onGoogleApiLoaded` event.

Everything else should go, including from `mocks/GoogleMaps` and `mocks/Mendix` files.